### PR TITLE
Catch `PredecessorTimelinePartWithoutEndException` and implemented `ApiErrors`

### PR DESCRIPTION
**Pull request by FredrikAreschoug**
fix for: https://sentry.io/organizations/hedvig/issues/1652079569/?environment=production&project=296779&referrer=alert_email

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/web/error/ApiError.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.hedvig.productPricing.service.web.error
+
+data class ApiError(
```
Comment (johantjellden) [inline src/main/java/com/hedvig/productPricing/service/web/error/ApiError.kt:3]: Should we start having a class like this returned when status code != 2XX?

Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/service/web/error/ApiError.kt:3]: I think that would be super nice, guessing we would want to have some internal error codes to be able to recover errors like this. Guessing Bacoffice just blows up in this case and recover that and present it to iex would be a lot nicer!

Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/service/web/error/ApiError.kt:3]: But I've been thinking and a builder for `ResponseEntity` would nice to implement for this!

File: src/main/java/com/hedvig/common/web/error/GlobalExceptionHandler.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  fun handleException(exception: PredecessorTimelinePartWithoutEndException, request: WebRequest): ResponseEntity<ApiErrorResponse> {
+    return ResponseEntity(
+      ApiErrorResponse(
+        ApiError(
+          code = predecessorTimelinePartWithoutEndCode,
+          message = predecessorTimelinePartWithoutEndMessage
+        )
+      ),
+      HttpHeaders(),
+      HttpStatus.CONFLICT
```
Comment (johantjellden) [inline src/main/java/com/hedvig/common/web/error/GlobalExceptionHandler.kt:24]: 👍

**Pull request merged**